core/src/main/scala/org/apache/spark/deploy/master/WorkerInfo.scala

```diff
@@ -101,5 +106,27 @@ private[spark] class WorkerInfo(
     this.state = state
   }
 
+  def setIsBlack(): Unit = {
+    this.isBlack = true
+    this.lastBlackTime = System.currentTimeMillis()
+  }
+
+  def unsetBlack(): Unit = {
+    this.isBlack = false
+    this.lastBlackTime = 0
+  }
+
+  def increaseFailedAppCount(appId: String) {
+    appIdToRetryCount(appId) = appIdToRetryCount.get(appId).map(_ + 1).getOrElse(1)
+  }
+
+  def getFailedFailedCount(appId: String): Int = {
+    appIdToRetryCount.getOrElse(appId, 1)
```

Comment:
@pgandhi999 Thank you for your review! Only when the number of times reaches the configured value “spark.deploy.executorFailedPerWorkerThreshold” , worker.isBlack will be set to true, otherwise it will not be affected.
When the application is finished, the information saved by appId in appIdToRetryCount will also be removed. So I didn't zero the number.